src/YouTrackSharp/Issues/Issue.cs

```diff
@@ -158,7 +163,9 @@ public override bool TrySetMember(SetMemberBinder binder, object value)
                     if (fieldElement.Value is JArray fieldElementAsArray)
                     {
                         // Map collection
-                        if (string.Equals(fieldElement.Name, "assignee", StringComparison.OrdinalIgnoreCase))
+                        // Heuristics for finding fields of the type List<Assignee>
+                        var children = new List<JToken>(fieldElementAsArray.First.Children());
+                        if(children.Count == 2 && children[0].Path == "[0].value" && children[1].Path == "[0].fullName")
```

Comment:
This seems like it may fail on various other types of objects or have unintended effects on other data types.

As an alternative, you could perhaps keep a property that explicitly contains all fields that are of the `Assignee` type? (and keep `"assignee"` as a default value for that property).

Reply:
Thank you for your comments. I agree with the sentiment, but I don't know how to do it -- in JSON there is no information about the field type, the JSON data looks like this:

````javascript
"issue":[
{
  "id":"PROJ-13",
  "entityId":"85-65385",
  "jiraId":null,
  "field":[
  ...,
  {
    "name":"Assignee",
    "value":[{"value":"jim.brown","fullName":"Jim Brown"}]
  },
  ...
}]
````
In YouTrack this file has type of `user`, however this information is not included in the data YouTrack sends to the client as JSON. Incidentally, when I run the same query in browser, I get response in XML and it _does_ include the information about the field type, but I don't think this is the kind of change we can make.

If you can give me any pointers as to how this detection could be made, I would be happy to implement it.

Reply:
I was thinking in `Issue.cs`, you could create a new property `AssigneeTypeFields` (or something with a better name), and then bring this line back:

``` 
if (string.Equals(fieldElement.Name, "assignee", StringComparison.OrdinalIgnoreCase))
```

Instead of checking for `"assignee"`, it would check `AssigneeTypeFields`.

However explaining this now made me realize this is not ideal when using YouTrackSharp. Will give it some more thought and see if there is a more reliable way of determining the type...

Reply:
Idea :-) For this line:
```
if(children.Count == 2 && children[0].Path == "[0].value" && children[1].Path == "[0].fullName")
```

Instead of checking things string-based using `Path`, could you check if the children are of type `JProperty`, and their names are correct?

Reply:
That would be great, thanks. However, the trick I am using is not so dangerous as it looks at the first glace -- you cannot make a user-defined field that has the same structure (i.e. the `value` is an object with two properties, "value" and "fullName"). Fields of the type `enum` have structure like this:

````javascript
{
  "name":"Department",
  "value":["3d Art"],
  "valueId":["3d Art"],
  "color":null
}
````

and similar structure is used for the fields of the type group, build and so on. So I agree it's inelegant, but I believe it's fairly safe.

Reply:
Previous comment was reply to your previous comment. About checking `JProperty`, I'll look into it.

Reply:
Did you mean something like:

````C#
if (children.Count == 2 && children[0] is JProperty && ((JProperty)children[0]).Name == "value"
  && children[1] is JProperty && ((JProperty)children[1]).Name == "fullName")
````

This works, should I change the pull request?

Reply:
That should do it, yes please!

Reply:
Done -- please check the latest commit.